conn.go

```diff
@@ -259,8 +263,10 @@ type Conn struct {
 	readErrCount  int
 	messageReader *messageReader // the current low-level reader
 
-	readDecompress         bool // whether last read frame had RSV1 set
-	newDecompressionReader func(io.Reader) io.ReadCloser
+	readDecompress         bool                          // whether last read frame had RSV1 set
+	newDecompressionReader func(io.Reader) io.ReadCloser // arges may flateReadWrapper struct
+
+	contextTakeover bool
```

Comment:
This field should not be required. All  connection logic for compression should be pushed into the the decompression reader and decompression writer.